compiler/rustc_trait_selection/src/traits/dyn_compatibility.rs

```diff
@@ -583,27 +583,29 @@ fn receiver_is_dispatchable<'tcx>(
     // create a modified param env, with `Self: Unsize<U>` and `U: Trait` (and all of
     // its supertraits) added to caller bounds. `U: ?Sized` is already implied here.
     let param_env = {
-        let param_env = tcx.param_env(method.def_id);
+        // N.B. We generally want to emulate the construction of the `unnormalized_param_env`
```

Comment:
So this currently misses out on the constness bounds which I guess is fine because associated type normalization doesn't care about constness? And also because `DispatchFromDyn` isnt a `const_trait` so it can only have non-const impls so again we don't really care about constness of stuff in the env.

We also miss out on the RPITIT projection preds? That's probable fine because it's not like we can define opaque types in this analysis anyway. I am assuming that the "assoc type" effectively has the item bounds of the opaque too so it won't result in being unable to prove stuff :woman_shrugging: 

Is that roughly correct? Can u add a small comment noting why we don't need to emulate those parts of param env construction

Reply:
Precisely correct. 

I could add the code to emulate those too but at that point I'd rather abstract this out into code shared by both the paramenv query and this, which is overkill imo (since it's still duplicated but slightly different for compare-impl-item).

I'll add a comment.